extensions/flight-sql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -0,0 +1,1710 @@
+        return error(Code.UNAUTHENTICATED, "Must be authenticated");
+    }
+
+    private static StatusRuntimeException permissionDeniedWithHelpfulMessage() {
+        return error(Code.PERMISSION_DENIED,
+                "Must use the original session; is the client echoing the authentication token properly? Some clients may need to explicitly enable cookie-based authentication with the header x-deephaven-auth-cookie-request=true (namely, Java Flight SQL JDBC drivers, and maybe others).");
+    }
+
+    private static StatusRuntimeException tableNotFound() {
+        return error(Code.NOT_FOUND, "table not found");
+    }
+
+    private static StatusRuntimeException transactionIdsNotSupported() {
+        return error(Code.INVALID_ARGUMENT, "transaction ids are not supported");
+    }
+
+    private static StatusRuntimeException queryParametersNotSupported(RuntimeException cause) {
+        return error(Code.INVALID_ARGUMENT, "query parameters are not supported", cause);
+    }
+
+    private class PreparedStatement {
+        private final ByteString handleId;
+        private final SessionState session;
+        private final String parameterizedQuery;
+        private final Set<CommandPreparedStatementQueryImpl> queries;
+        private final Closeable onSessionClosedCallback;
+
+        PreparedStatement(SessionState session, String parameterizedQuery) {
+            this.session = Objects.requireNonNull(session);
+            this.parameterizedQuery = Objects.requireNonNull(parameterizedQuery);
+            this.handleId = ByteString.copyFromUtf8(UUID.randomUUID().toString());
+            this.queries = new HashSet<>();
+            preparedStatements.put(handleId, this);
+            this.session.addOnCloseCallback(onSessionClosedCallback = this::onSessionClosed);
+        }
+
+        public ByteString handleId() {
+            return handleId;
+        }
+
+        public String parameterizedQuery() {
+            return parameterizedQuery;
+        }
+
+        public void verifyOwner(SessionState session) {
+            if (!this.session.equals(session)) {
+                // We should not be concerned about returning "NOT_FOUND" here; the handleId is sufficiently random that
+                // it is much more likely an authentication setup issue.
+                throw permissionDeniedWithHelpfulMessage();
+            }
+        }
+
+        public synchronized void attach(CommandPreparedStatementQueryImpl query) {
+            queries.add(query);
+        }
+
+        public synchronized void detach(CommandPreparedStatementQueryImpl query) {
+            queries.remove(query);
+        }
+
+        public void close() {
+            closeImpl();
+            session.removeOnCloseCallback(onSessionClosedCallback);
+        }
+
+        private void onSessionClosed() {
+            log.debug().append("onSessionClosed: removing prepared statement ").append(handleId.toString()).endl();
```

Comment:
nitpicky, so ignore if you want (but I think this might be the kind of thing you'd flag if you spotted it):
This is a little gross since this is the utf8 of a uuid string (36 bytes), and we pay for the toString() of the ByteString whether or not we log the debug... 

Technically I think handleId.asReadOnlyByteBuffer would at least not copy it every time, and the bytebuffer is holding ascii instead of the actual bytes of the uuid, so that would print nicely too.

But do we make handleId by design to hold ascii? Could use UuidCreator.toBytes(UUID) and have 16 real bytes instead of 8 smeared across the LSBs of 32ish bytes? At that point you might consider just using SecureRandom.nextBytes(16) since we don't actually use the uuid api at all past this point.

Also worth noting that the length-36 ByteStrings  going to get hashed by multiplying each by 31 and adding the next... my math isn't great here, but packing so many zeros isn't going to make for nice hashes right. Won't matter except under load, but then it might matter if we are collision heavy.

Could easily wait until it does matter, but it seems cheap to either extract the bytes (or generate the bytes) and live with ByteString's toString and wrap in an if debug check?